### PR TITLE
#5698 Fix mod options on marketplace activation with personal DB creation

### LINKED ISSUE #5698: Regression: mod inputs not showing during marketplace activation
Context
---
* https://app.rainforestqa.com/runs/1331141/tests/379021/browsers/windows10_chrome
* https://www.pixiebrix.com/marketplace/cc1315aa-4939-4e81-ac76-ca60689891b9/highlight-specific-keywords-when-page-loads/
* Seems to be impacting all mods

Potentially Related Failure
---
- https://app.rainforestqa.com/runs/1331287/tests/378050/browsers/windows10_chrome?source=email

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Fixes #5698 

## Reviewer Tips

- Sorry, my rename of the `useWizard` hook kinda nuked the diff on this PR 🤦  see comments for changes

## Checklist

- [ ] Add tests - Already have coverage + I'll be adding rainforest tests to cover the DB activation stuff
- [x] Designate a primary reviewer - @twschiller

### CODE REVIEW

File: src/activation/useActivateRecipeWizard.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  optionSchema: Schema,
+  name: string
+): string {
+  return `${recipe.metadata.name} - ${optionSchema.title ?? name}`;
+}
+
+function useActivateRecipeWizard(
+  recipe: RecipeDefinition,
+  defaultAuthOptions: Record<RegistryId, AuthOption> = {}
+): FetchableAsyncState<UseActivateRecipeWizardResult> {
```
Comment (BLoe) [inline src/activation/useActivateRecipeWizard.ts:79]: Hook now returns `FetchableAsyncState`

File: src/activation/useActivateRecipeWizard.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  const installedExtensions = useSelector(selectExtensions);
+  const optionsValidationSchemaState = useAsyncRecipeOptionsValidationSchema(
+    recipe.options?.schema
+  );
+  const databaseOptionsState = useDatabaseOptions();
```
Comment (BLoe) [inline src/activation/useActivateRecipeWizard.ts:84]: This hook needs to load database options now. Both hooks called here are refactored to return `FetchableAsyncState`, merged below.

File: src/activation/useActivateRecipeWizard.test.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+import useActivateRecipeWizard, {
+  makeDatabasePreviewName,
+} from "@/activation/useActivateRecipeWizard";
+import { renderHook } from "@testing-library/react-hooks";
+import { propertiesToSchema } from "@/validators/generic";
+import useDatabaseOptions from "@/hooks/useDatabaseOptions";
+import { valueToAsyncState } from "@/utils/asyncStateUtils";
+
+jest.mock("@/components/auth/AuthWidget", () => {});
+jest.mock("react-redux");
```
Comment (twschiller) [inline src/activation/useActivateRecipeWizard.test.tsx:29]: 😞

File: src/activation/useActivateRecipeWizard.ts [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+                optionSchema,
+                name
+              );
+              const existingDatabaseOption = databaseOptions.find(
+                (option) => option.label === `${databaseName} - Private`
+              );
+              return existingDatabaseOption?.value ?? databaseName;
+            }
+
+            return forcePrimitive(optionSchema.default);
```
Comment (BLoe) [inline src/activation/useActivateRecipeWizard.ts:157]: The entire memo body was moved into the merge state callback, and the only change is here. This handles existing auto-created database by matching the name to a loaded database option. The former code for the mapping function here was:
```
(optionSchema: Schema, name: string) => {
  const installed = installedOptions[name];
  if (installed) {
    return forcePrimitive(installed);
  }

  if (
    isDatabaseField(optionSchema) &&
    optionSchema.format === "preview"
  ) {
    return `${blueprint.metadata.name} - ${optionSchema.title ?? name}`;
  }

  return forcePrimitive(optionSchema.default);
}
```

File: src/components/fields/schemaFields/widgets/DatabaseWidget.tsx [hunk 5]
```diff
@@ -45,23 +44,21 @@ const DatabaseWidget: React.FunctionComponent<SchemaFieldProps> = ({
   >(name);
   const { allowExpressions } = useContext(FieldRuntimeContext);
 
-  const { databaseOptions, isLoading: isLoadingDatabaseOptions } =
+  const { data: databaseOptions, isLoading: isLoadingDatabaseOptions } =
```
Comment (twschiller) [inline src/components/fields/schemaFields/widgets/DatabaseWidget.tsx:47]: @BLoe now that useDatabaseOptions fetches on mount. There will be 2 requests - once for the hook, and once to render the widget. Is that OK?

File: src/hooks/useDatabaseOptions.ts [hunk 6]
```diff
@@ -16,35 +16,35 @@
...
-    useGetDatabasesQuery();
-  const { data: organizations, isLoading: isLoadingOrganizations } =
-    useGetOrganizationsQuery();
+const useDatabaseOptions = (): FetchableAsyncState<Option[]> => {
+  const databasesQueryState = useGetDatabasesQuery(undefined, {
+    refetchOnMountOrArgChange: true,
+  });
+  const organizationsQueryState = useGetOrganizationsQuery(undefined, {
+    refetchOnMountOrArgChange: true,
+  });
```
Comment (BLoe) [inline src/hooks/useDatabaseOptions.ts:30]: Making sure we have up-to-date data at the call sites for this hook

**Review by twschiller — APPROVED** [review]
@BLoe see note here: https://github.com/pixiebrix/pixiebrix-extension/pull/5709/files#r1190328311

I think this is OK to get out for QA cycle though

**Pull request merged**